doc/release/0.17.0-notes.rst

```diff
@@ -29,6 +29,9 @@ Deprecated features
 Backwards incompatible changes
 ==============================
 
+The functions ``trim1`` and ``trimboth`` now sort elements before trimming.
+This is backwards incompatible but it made no sense to trim unsorted input.
```

Comment:
full sorting is not necessary, a partial sort is enough, see `np.partition` available since 1.8. As a fallback you can use a full sort, but I wouldn't document the result to be fully sorted.